### PR TITLE
:sparkles: Allow searching repo by DID

### PR DESCRIPTION
**Pull request by foysalit**
This PR allows `admin.searchRepos` to take `did` as a search term along with handles.

### CODE REVIEW

**Comment by devinivy** [comment]
This seems like a reasonable approach to me 👍 I like that this does not have to affect the behavior of `accountService.search()` other places where it's used.

**Comment by foysalit** [comment]
@devinivy ported over to bsky package along with tests. believe it's ready for review.

File: packages/bsky/src/api/com/atproto/admin/searchRepos.ts [hunk 1]
```diff
@@ -16,9 +17,17 @@ export default function (server: Server, ctx: AppContext) {
         throw new InvalidRequestError('The invitedBy parameter is unsupported')
       }
 
+      let searchField: 'did' | 'handle'
+      try {
+        ensureValidDid(term)
+        searchField = 'did'
+      } catch (err) {
+        searchField = 'handle'
+      }
```
Comment (Matrix89) [inline packages/bsky/src/api/com/atproto/admin/searchRepos.ts:26]: Throwing and then immediately catching is a bit of an anti-pattern IMO(hidden control flow, error creation overhead etc.), but there's no `isValidDid` method so I guess it's ok

Comment (dholms) [reply packages/bsky/src/api/com/atproto/admin/searchRepos.ts:26]: Generally we handle this case with:

`switchField = term.startWith('did:') ? 'did' : 'handle'`

File: packages/pds/src/services/account/index.ts [hunk 2]
```diff
@@ -333,11 +333,25 @@ export class AccountService {
   }
 
   async search(opts: {
+    searchField?: 'did' | 'handle'
     term: string
     limit: number
     cursor?: string
     includeSoftDeleted?: boolean
   }): Promise<(RepoRoot & DidHandle & { distance: number })[]> {
+    // TODO: Implement sqlite specific version?
```
Comment (dholms) [inline packages/pds/src/services/account/index.ts:342]: I don't see anything sqlite specific here

File: packages/bsky/tests/views/admin/repo-search.test.ts
Comment (dholms) [inline packages/bsky/tests/views/admin/repo-search.test.ts:1]: feels a bit weird to have this one admin test here & none of the others. but i guess it doesn't hurt 😅

File: packages/bsky/src/api/com/atproto/admin/searchRepos.ts (hunk 1 above)
Comment (foysalit) [reply packages/bsky/src/api/com/atproto/admin/searchRepos.ts:26]: Felt safer to use `ensureDid` but sure, I can replace this with the `startWith()` approach.

**Comment by foysalit** [comment]
@dholms @devinivy addressed all comments and made necessary adjustments, I'd really appreciate another round of review and hopefully a ✅

**Review by devinivy — APPROVED** [review]
Looking good!

**Review by devinivy — APPROVED** [review]
Looking good!

**Comment by bnewbold** [comment]
I think this is good to merge, and I could push the button? Don't want this to get stale and out of sync with `main`.

**Comment by devinivy** [comment]
@bnewbold thanks for the ping 👍 merged!

**Pull request merged**